File: 20181123-java-leap-year/src/test/java/com/katas/LeapYearTest.java
package com.katas;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class LeapYearTest {
    @Test
    public void should_be_leap_year_when_divisible_by_4() {
        boolean result = isLeapYear(1996);

        assertThat(result).isTrue();
    }

    @Test
    public void should_not_be_leap_year_when_non_divisible_by_4() {
        boolean result = isLeapYear(1997);

        assertThat(result).isFalse();
    }

    @Test
    public void should_not_be_leap_year_when_divisible_by_4_and_also_100() {
        boolean result = isLeapYear(1900);

        assertThat(result).isFalse();
    }

    @Test
    public void should_be_leap_year_when_divisible_by_4_and_also_400() {
        boolean result = isLeapYear(2000);

        assertThat(result).isTrue();
    }

    private boolean isLeapYear(int year) {
        return year % 4 == 0
                && year % 100 != 0
                || year % 400 == 0;
    }
}

File: 20210320-kotlin-leapyear/src/test/kotlin/com/katas/LeapYearTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class LeapYearTest {
    @Test
    internal fun `should be leap year if divisible by 4`() {
        assertThat(isLeapYear(1996)).isTrue()
    }

    @Test
    internal fun `should be common year if not divisible by 4`() {
        assertThat(isLeapYear(2001)).isFalse()
    }

    @Test
    internal fun `should be atypical common year if divisible by 4 and 100`() {
        assertThat(isLeapYear(1900)).isFalse()
    }
}
